front/src/app/js/controllers/users/calendar.js

```diff
@@ -1,47 +1,85 @@
+import moment from 'moment';
+import {API_PREFIX} from 'upont/js/config/constants';
+
 /* @ngInject */
 class Calendar_Ctrl {
-    constructor($rootScope, $scope, $filter, events, courseItems, calendarConfig) {
-        $scope.events = [];
+    constructor($http, $scope, calendar, calendarConfig) {
+        $scope.loading = false;
 
-        // FIXME
-        events = events.data;
+        $scope.toScope = (calendar) => {
+            console.log(calendar);
+            $scope.calendarEvents = calendar.events;
+            $scope.calendarView = calendar.view;
+            $scope.calendarDay = calendar.day;
+        };
+        $scope.toScope(calendar);
 
-        for (let i = 0; i < events.length; i++) {
-            let type;
-            switch (events[i].entry_method) {
-            case 'Shotgun':
-                type = 'important';
-                break;
-            case 'Libre':
-                type = 'warning';
-                break;
-            case 'Ferie':
-                type = 'success';
-                break;
+        const updateView = () => {
+            if ($scope.loading) {
+                return;
             }
-            $scope.events.push({
-                color: calendarConfig.colorTypes[type],
-                startsAt: new Date(events[i].start_date * 1000),
-                endsAt: new Date(events[i].end_date * 1000),
-                title: events[i].author_club.name + ' : ' + events[i].name,
-                editable: false,
-                deletable: false,
-                draggable: false,
-                resizable: false,
-                incrementsBadgeTotal: true
+
+            $scope.loading = true;
+            Calendar_Ctrl.getCalendar($http, calendarConfig, $scope.calendarView, $scope.calendarDay).then(
+                (calendar) => $scope.toScope(calendar),
+                () => console.error('Failed to update calendar view'),
+            ).finally(() => {
+                $scope.loading = false;
             });
         }
 
-        for (let i = 0; i < courseItems.length; i++) {
-            const group = courseItems[i].group;
-            $scope.events.push({
-                color: calendarConfig.colorTypes.info,
-                startsAt: new Date(courseItems[i].start_date * 1000),
-                endsAt: new Date(courseItems[i].end_date * 1000),
-                title: '[' + courseItems[i].location + '] ' + courseItems[i].course.name + (
-                    group !== 0
-                        ? ' (Gr ' + group + ')'
-                        : ''),
+        $scope.$watchGroup(
```

Comment:
Tu fais quoi avec tes $watchGroup ?

Reply:
je fix `// If any of these scope variables change, update the view by fetching events from the API`